pkg/agent/flowexporter/connections/conntrack_linux_test.go

```diff
@@ -101,19 +113,85 @@ func TestConnTrackSystem_DumpFlows(t *testing.T) {
 		GatewayConfig: gwConfig,
 		PodIPv4CIDR:   podCIDR,
 	}
-	// Test the DumpFlows implementation of connTrackSystem
-	mockNetlinkCT := connectionstest.NewMockNetFilterConnTrack(ctrl)
-	connDumperDPSystem := NewConnTrackSystem(nodeConfig, svcCIDR, netip.Prefix{}, false)
+	testCases := []struct {
+		name                string
+		protocols           []string
+		testFlows           []*flowexporter.Connection
+		expectedConnections int
+	}{
+		{
+			"Filter unhandled flow types",
+			nil,
+			testVarietyFlows,
+			1,
+		},
+		{
+			"Filter for all supported protocols with explicit declaration",
+			[]string{"tcp", "udp", "sctp"},
+			testFlowsMixedProtocols,
+			6,
+		},
+		{
+			"Filter for all supported protocols with default config",
+			nil,
+			testFlowsMixedProtocols,
+			6,
+		},
+		{
+			"Filter for only TCP flows",
+			[]string{"tcp"},
+			testFlowsMixedProtocols,
+			3,
+		},
+		{
+			"Filter for only UDP",
+			[]string{"udp"},
+			testFlowsMixedProtocols,
+			2,
+		},
+		{
+			"Filter for only SCTP",
+			[]string{"sctp"},
+			testFlowsMixedProtocols,
+			1,
+		},
+		{
+			"Filter for only TCP and SCTP",
+			[]string{"tcp", "sctp"},
+			testFlowsMixedProtocols,
+			4,
+		},
+		{
+			"Filter for UDP and SCTP",
+			[]string{"udp", "sctp"},
+			testFlowsMixedProtocols,
+			3,
+		},
```

Comment:
Honestly, I would just keep one of thee here (maybe "filter for only TCP and UDP" which is how it will typically be used). The `protocolFilter` implementation is already unit-tested independently.